sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/SyntheticDataPubSubPublisher.java

```diff
@@ -83,14 +95,41 @@ public static void main(String[] args) throws IOException {
 
     Pipeline pipeline = Pipeline.create(options);
 
-    pipeline
-        .apply("Read synthetic data", Read.from(new SyntheticBoundedSource(sourceOptions)))
-        .apply("Map to PubSub messages", MapElements.via(new MapBytesToPubSubMessage()))
-        .apply("Write to PubSub", PubsubIO.writeMessages().to(options.getInsertionPipelineTopic()));
-
+    if (options.getKafkaBootstrapServerAddress() != null) {
```

Comment:
Personally, I would recommend to split those two pipelines into separate files. 

@lgajowy WDYT?

Reply:
IMO it can stay as one file. After all, we can have multiple sinks in one pipeline and even publish to both at the same time if we want to (no idea that this is ever going to be a case, though...  :) ) . I'd split it like I suggested in the review comments below.